ospd/vtfilter.py

```diff
@@ -128,4 +128,4 @@ def get_filtered_vts_list(self, vts, vt_filter):
                 else:
                     _vts_aux.pop(vt_id)
 
-        return _vts_aux
+        return _vts_aux if len(_vts_aux) else -1
```

Comment:
I had a look at the return value docstring and it says "Dictionary with filtered vulnerability tests.". In that case it would be better to return None here to indicate an error/special case. Also the docstring should be updated to reflect this change.
```suggestion
        return _vts_aux if len(_vts_aux) else None
```

Reply:
If you don't use a filter, the default filter is None. Then, instead of getting the complete VTs set , you get none.
We need to differentiate the "no filter" with the "no matches". A no matches returned an empty dictionary, and this was the initial problem, so when the filter does not have matches, it returned the complete set.

Reply:
Ok the None case when no filter isn't mentioned in the docs too. Therefore I did oversee it in the function code. But I still don't understand why you want to return -1 when the dict is empty instead of checking at the place where this is used for an empty dict as in my suggestion here https://github.com/greenbone/ospd/pull/165/files/05918e21c73c861ea88c3064ff0e6d3a555e2130#r339015657